### ISSUE TITLE
Non random crash when evaluating AI actions

### ISSUE THREAD

**Issue by xaba** [wesnoth/wesnoth#1798]
Wesnoth 1.13.8 crashes when examining AI's units actions.
```gdb
Thread 1 "wesnoth" received signal SIGSEGV, Segmentation fault.
wfl::variant_list::deref_iterator (this=<optimized out>, iter=...) at ./src/formula/variant_value.cpp:345
(gdb) bt
#0  wfl::variant_list::deref_iterator (this=<optimized out>, iter=...) at ./src/formula/variant_value.cpp:345
#1  0x0000555555f1041e in wfl::variant_iterator::operator* (this=this@entry=0x7fffffffafd0) at ./src/formula/variant.cpp:83
#2  0x0000555555effd8e in wfl::builtins::min_function::execute (this=0x55555a47cf10, variables=..., fdb=0x0) at ./src/formula/function.cpp:205
#3  0x0000555555db7111 in wfl::formula_expression::evaluate (this=0x55555a47cf10, variables=..., fdb=0x0) at ./src/formula/function.hpp:41
#4  0x0000555555ee0099 in wfl::operator_expression::execute (this=0x55555a01f2c0, variables=..., fdb=<optimized out>) at ./src/formula/formula.cpp:708
#5  0x0000555555db7111 in wfl::formula_expression::evaluate (this=0x55555a01f2c0, variables=..., fdb=0x0) at ./src/formula/function.hpp:41
#6  0x0000555555ee0034 in wfl::operator_expression::execute (this=0x55555a01f650, variables=..., fdb=0x0) at ./src/formula/formula.cpp:707
#7  0x0000555555db7111 in wfl::formula_expression::evaluate (this=0x55555a01f650, variables=..., fdb=0x0) at ./src/formula/function.hpp:41
#8  0x0000555555ee0034 in wfl::operator_expression::execute (this=0x55555a26d700, variables=..., fdb=0x0) at ./src/formula/formula.cpp:707
#9  0x0000555555db7111 in wfl::formula_expression::evaluate (this=0x55555a26d700, variables=..., fdb=0x0) at ./src/formula/function.hpp:41
#10 0x0000555555ed7a61 in wfl::formula::execute (this=<optimized out>, variables=..., fdb=fdb@entry=0x0) at ./src/formula/formula.cpp:234
#11 0x0000555555d93a0b in wfl::formula::evaluate (fdb=0x0, variables=..., this=<optimized out>) at ./src/formula/formula.hpp:59
#12 wfl::formula::evaluate (fdb=0x0, default_res=<error reading variable: access outside bounds of object referenced via synthetic pointer>, variables=..., 
    f=std::shared_ptr (count 1508218176, weak 21844) 0x7fffffffb730) at ./src/formula/formula.hpp:46
#13 wfl::base_candidate_action::execute_formula (formula=std::shared_ptr (count 1, weak 0) 0x55555b4c2c20, callable=..., ai=ai@entry=0x55555a118330, this=0x55555a9d9028)
    at ./src/ai/formula/candidates.cpp:47
#14 0x0000555555d97808 in wfl::attack_candidate_action::evaluate (this=0x55555a9d8fe0, ai=0x55555a118330, units=...) at ./src/ai/formula/candidates.cpp:234
#15 0x0000555555d9c773 in ai::fai_candidate_action_wrapper::evaluate (this=0x55555a576f60) at ./src/ai/formula/engine_fai.cpp:56
#16 0x0000555555d79925 in ai::ai_default_rca::candidate_action_evaluation_loop::do_play_stage (this=0x55555aa50400) at ./src/ai/default/stage_rca.cpp:112
#17 0x0000555555ce9234 in ai::ai_composite::play_turn (this=this@entry=0x55555aa50350) at ./src/ai/composite/ai.cpp:150
#18 0x0000555555dd8510 in ai::manager::play_turn (side=5) at ./src/ai/manager.cpp:776
#19 0x0000555556350aa2 in playsingle_controller::play_ai_turn (this=this@entry=0x7fffffffc900) at ./src/playsingle_controller.cpp:555
#20 0x0000555556350f68 in playsingle_controller::play_side_impl (this=0x7fffffffc900) at ./src/playsingle_controller.cpp:397
#21 0x0000555556349e05 in play_controller::play_side (this=this@entry=0x7fffffffc900) at ./src/play_controller.cpp:1137
#22 0x000055555634a876 in play_controller::play_turn (this=0x7fffffffc900) at ./src/play_controller.cpp:1175
#23 0x0000555556351fc6 in playsingle_controller::play_scenario_main_loop (this=this@entry=0x7fffffffc900) at ./src/playsingle_controller.cpp:181
#24 0x0000555556353649 in playsingle_controller::play_scenario (this=this@entry=0x7fffffffc900, level=...) at ./src/playsingle_controller.cpp:270
#25 0x0000555555fbc8bf in campaign_controller::playsingle_scenario (this=this@entry=0x7fffffffcf20, end_level=...) at ./src/game_initialization/playcampaign.cpp:194
#26 0x0000555555fbcd23 in campaign_controller::play_game (this=this@entry=0x7fffffffcf20) at ./src/game_initialization/playcampaign.cpp:281
#27 0x0000555555fcac37 in game_launcher::launch_game (this=0x555556e81980, reload=<optimized out>) at ./src/game_launcher.cpp:953
#28 0x0000555555c41883 in do_gameloop (args=std::vector of length 1, capacity 1 = {...}, this=<optimized out>, this=<optimized out>, this=<optimized out>) at ./src/wesnoth.cpp:827
#29 main (argc=<optimized out>, argv=<optimized out>) at ./src/wesnoth.cpp:1055
```
Is this bug the same as https://github.com/wesnoth/wesnoth/issues/1763?
The AI's units also have a poisoning attack in my case.
I can provide the saved game in case it's needed.

**Comment by CelticMinstrel** [comment]
From the stack trace, it does look likely the same as #1763.

**Comment by CelticMinstrel** [comment]
Presumably fixed in fd1f115.

**Comment by CelticMinstrel** [comment]
Presumably fixed in fd9f115.
